mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1553,8 +1714,29 @@ createCPUConvWithMLIR(ModuleOp module, func::FuncOp &func,
   affine::buildAffineLoopNest(b, loc, lowerBounds, upperBounds, steps,
                               createConv2dLoopNest);
 
+  if (!opd1.dyn_cast<BlockArgument>())
```

Comment:
Undo the F32-buffer copies at the end -- dealloc for arguments, copy back to blockArgument for the result.